src/main/java/org/springframework/cloud/bindings/boot/EurekaBindingsPropertiesProcessor.java

```diff
@@ -47,9 +55,82 @@ public void process(Environment environment, Bindings bindings, Map<String, Obje
             map.from("uri").to("eureka.client.serviceUrl.defaultZone",
                     (uri) -> String.format("%s/eureka/", uri)
             );
-
             properties.put("eureka.client.region", "default");
-        });
 
+            Map<String, String> secret = binding.getSecret();
+            String caCert = secret.get("ca.crt");
+            if (caCert != null && !caCert.isEmpty()) {
+                // generally apps using TLS bindings will be running in k8s where the host name is not meaningful,
+                // but we don't want to override the endpoint behavior the app has already set, in case they want to
+                // explicitly set eureka.instance.hostname to route traffic through normal ingress.
+                if (! environment.containsProperty("eureka.instance.preferIpAddress")) {
+                    properties.put("eureka.instance.preferIpAddress", true);
+                }
+
+                Random random = new Random();
+                String generatedPassword = random.ints(97 /* letter a */, 122 /* letter z */ + 1)
+                        .limit(10)
+                        .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+                        .toString();
+
+                // Create a trust store from the CA cert
+                String trustFilePath = Paths.get(System.getProperty("java.io.tmpdir"), "client-truststore.p12").toString();
+                try {
+                    KeyStore ks = PemSslStoreHelper.createKeyStore("trust", "PKCS12", caCert, null, "rootca");
+                    FileOutputStream fos = new FileOutputStream(trustFilePath);
+                    ks.store(fos, generatedPassword.toCharArray());
+                    fos.close();
+                } catch (FileNotFoundException e) {
+                    throw new IllegalStateException("Unable to open truststore output file", e);
+                } catch (KeyStoreException e) {
+                    throw new IllegalStateException("Unable to write truststore", e);
+                } catch (NoSuchAlgorithmException e) {
+                    throw new IllegalStateException("Unable to write truststore", e);
+                } catch (CertificateException e) {
+                    throw new IllegalStateException("Unable to process certificate", e);
+                } catch (IOException e) {
+                    throw new IllegalStateException("Unable to create truststore", e);
+                }
+                properties.put("eureka.client.tls.enabled", true);
+                properties.put("eureka.client.tls.trust-store", "file:"+trustFilePath);
+                properties.put("eureka.client.tls.trust-store-type", "PKCS12");
+                properties.put("eureka.client.tls.trust-store-password", generatedPassword);
+
+                // When tls.crt and tls.key are set, enable mTLS for Eureka
+                String clientKey = secret.get("tls.key");
```

Comment:
this code is a duplicate of what you wrote above; please extract in a common method, like this:

```java
    private static void createStore(String generatedPassword, String filePath, KeyStore ks, String storeType) {
        try {
            FileOutputStream fos = new FileOutputStream(filePath);
            try {
                ks.store(fos, generatedPassword.toCharArray());
            } catch (KeyStoreException e) {
                throw new IllegalStateException("Unable to write " + storeType, e);
            } catch (NoSuchAlgorithmException e) {
                throw new IllegalStateException("Cryptographic algorithm not available", e);
            } catch (CertificateException e) {
                throw new IllegalStateException("Unable to process certificate", e);
            } catch (IOException e) {
                throw new IllegalStateException("Unable to create " + storeType, e);
            } finally {
                try {
                    fos.close();
                } catch (IOException e) {
                    throw new IllegalStateException("Unable to close " + storeType + " output file", e);
                }
            }
        } catch (FileNotFoundException e) {
            throw new IllegalStateException("Unable to open " + storeType + " output file", e);
        }
    }
```